crates/bevy_pbr/src/meshlet/persistent_buffer.rs

```diff
@@ -110,17 +109,18 @@ impl<T: PersistentGpuBufferable> PersistentGpuBuffer<T> {
 }
 
 /// A trait representing data that can be written to a [`PersistentGpuBuffer`].
-///
-/// # Safety
-/// * All data must be a multiple of `wgpu::COPY_BUFFER_ALIGNMENT` bytes.
-/// * The amount of bytes written to `buffer` in `write_bytes_le()` must match `size_in_bytes()`.
-pub unsafe trait PersistentGpuBufferable {
+pub trait PersistentGpuBufferable {
     /// Additional metadata associated with each item, made available during `write_bytes_le`.
     type Metadata;
 
-    /// The size in bytes of `self`.
+    /// The size in bytes of `self`. This will determine the size of the buffer passed into
+    /// `write_bytes_le`.
+    ///
+    /// All data written must be in a multiple of `wgpu::COPY_BUFFER_ALIGNMENT` bytes. Failure to do so will
+    /// result in a panic when using [`PersistentGpuBuffer`].
     fn size_in_bytes(&self) -> usize;
 
     /// Convert `self` + `metadata` into bytes (little-endian), and write to the provided buffer slice.
+    /// Any bytes not written to in the slice will be zeroed out when uploaded to the GPU.
```

Comment:
Theoretically, it would be nice for wgpu to provide a way to skip the zeroing, so this may change in the future if they add that. We can leave it for now though :)

Reply:
I asked about this in the wgpu matrix chat and it seems feasible, though it may need to be exposed via &mut [MaybeUninit<u8>], which would only require unsafe inside the implementation instead of as a part of the trait itself.